platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultBuildTreeModelControllerServices.kt

```diff
@@ -74,8 +74,12 @@ class DefaultBuildTreeModelControllerServices : BuildTreeModelControllerServices
          */
         private
         val modelProjectDependencies = InternalFlag("org.gradle.internal.model-project-dependencies", true)
+
+        private
+        val configurationCacheForModels = InternalFlag("org.gradle.configuration-cache.internal.tooling-models", false)
```

Comment:
Initially, we introduce this feature behind a flag due to a potential high exposure. Assuming a high number of user builds having the Configuration Cache property enabled, not guarding this change with a flag would result in it being unconditionally applicable to all IDE syncs in such builds.

At a later point, when we have some feedback from early users, we should enable this by default.